docs/src/developers_guide/release.rst

```diff
@@ -61,73 +126,93 @@ for example:
-cut, a new pre-release shall be issued first.
+Tagging
+~~~~~~~
 
-Make the release candidate available as a conda package on the
-`conda-forge Anaconda channel`_ using the `rc_iris`_ label. To do this visit
-the `conda-forge iris-feedstock`_ and follow `CFEP-05`_. For further information
-see the `conda-forge User Documentation`_.
+Once all checks are complete, the release is published from the release
+branch - via the GitHub release functionality in the ``SciTools/iris``
+repository - which simultaneously creates a Git tag for the release.
 
 
-Documentation
--------------
+Post Release
+------------
 
-The documentation should include all of the ``whatsnew`` entries for the release.
-This content should be reviewed and adapted as required.
+PyPI
+~~~~
+Iris is available on PyPI as ``scitools-iris``.
 
-Steps to achieve this can be found in the :ref:`iris_development_releases_steps`.
+Iris' Continuous-Integration (CI) includes the automatic building and publishing of
+PyPI artifacts in a dedicated GitHub Action.
 
+Legacy manual instructions are appended to this page for reference purposes
+(:ref:`update_pypi`)
 
-The Release
------------
+conda-forge
+~~~~~~~~~~~
+
+Iris is available on conda-forge as ``iris``.
 
-The final steps of the release are to ensure that the release date and details
-are correct in the relevant ``whatsnew`` page within the documentation.
+This is managed via the the Iris conda recipe on the
+`conda-forge iris-feedstock`_, which is updated after the release is cut on
+GitHub, followed by automatic build and publish of the
+conda package on the `conda-forge Anaconda channel`_.
 
-There is no need to update the ``iris.__version__``, as this is managed
-automatically by `setuptools-scm`_.
+Announcement
+~~~~~~~~~~~~
 
-Once all checks are complete, the release is published on GitHub by
-creating a new tag in the ``SciTools/iris`` repository.
+Iris uses Twitter (`@scitools_iris`_) to announce new releases, as well as any
+internal message boards that are accessible (e.g. at the UK Met Office).
+Announcements usually include a highlighted feature to hook readers' attention.
 
+Citation
+~~~~~~~~
 
-Update conda-forge
-------------------
+``docs/src/userguide/citation.rst`` is updated to include
+the latest [non-release-candidate] version, date and `Zenodo DOI`_
+of the new release. Ideally this would be updated before the release, but
+the DOI for the new version is only available once the release has been
+created in GitHub.
+
+Merge Back
+~~~~~~~~~~
+
+After any release is published, **including patch releases**, the changes from the
+release branch should be merged back onto the ``SciTools/iris`` ``main`` branch.
 
-Once a release is cut on GitHub, update the Iris conda recipe on the
-`conda-forge iris-feedstock`_ for the release. This will build and publish the
-conda package on the `conda-forge Anaconda channel`_.
 
+Appendices
+----------
 
 .. _update_pypi:
 
-Update PyPI
------------
+Updating PyPI Manually
```

Comment:
This section needs updating, as this method of manual update didn't seem to work

Reply:
Agreed! Looks like @bjlittle has something for this in #5252, so assuming that gets merged I can bring that into this PR.